### PR TITLE
Add unit tests for metabolizing of all reagents and fix the ones that failed

### PR DESCRIPTION
**Pull request by Mothblocks**
## About The Pull Request
Adds a unit test that goes through every reagent and ensures that it metabolizes on both a human and a monkey without runtiming.

## Why It's Good For The Game
Helps prevent future creation of reagents that assume too much, like that all consumers will have minds or will be humans.

## Changelog
:cl:
fix: Fixed several edge cases where some reagents would prevent metabolization.
/:cl:

### CODE REVIEW

**Comment by Jared-Fogle** [comment]
There's problematic cases where a lot of chemicals only do things based on chance. Ideally `prob` could be mocked out to test this, though I'm not including that in this PR.

File: code/modules/mob/living/carbon/carbon.dm [hunk 1]
```diff
@@ -506,7 +506,7 @@
 			if(T)
 				T.add_vomit_floor(src, VOMIT_TOXIC, purge)//toxic barf looks different || call purge when doing detoxicfication to pump more chems out of the stomach.
 		T = get_step(T, dir)
-		if (is_blocked_turf(T))
+		if (T != null && is_blocked_turf(T))
```
Comment (ShizCalev) [inline code/modules/mob/living/carbon/carbon.dm:509]: this should be fixed in is_blocked_turf instead

Comment (ShizCalev) [inline code/modules/mob/living/carbon/carbon.dm:509]: is_blocked_turf should be fixed instead

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/carbon.dm (hunk 1 above)
Comment (Jared-Fogle) [reply code/modules/mob/living/carbon/carbon.dm:509]: To accept null turfs?

**Pull request merged**